lib/bolt/applicator.rb

```diff
@@ -48,12 +56,20 @@ def custom_facts_task
       end
     end
 
-    def catalog_apply_task
+    def catalog_apply_task(catalog_path, plugins_path)
       @catalog_apply_task ||= begin
         path = File.join(libexec, 'apply_catalog.rb')
-        file = { 'name' => 'apply_catalog.rb', 'path' => path }
-        metadata = { 'supports_noop' => true, 'input_method' => 'stdin' }
-        Bolt::Task.new(name: 'apply_helpers::apply_catalog', files: [file], metadata: metadata)
+        files = [
+          { 'name' => 'apply_catalog.rb', 'path' => path },
+          { 'name' => 'catalog.json', 'path' => catalog_path },
```

Comment:
I think we should keep the catalog as a parameter to the task rather then a file since it will be different for every invocation.

Reply:
might be smart to compress the catalog since it's JSON by default, eventually we may run into a spot where the catalog is large (>10MB) and that will cause the same failures